utils.go

```diff
@@ -0,0 +1,53 @@
+package dpop
+
+import (
+	"crypto"
+	"encoding/base64"
+)
+
+// Internal function used to ensure hash is available and set the default
+// to SHA256
+func ValidateHashFunction(args ...crypto.Hash) (*crypto.Hash, error) {
+	hashFn := new(crypto.Hash)
+	if len(args) > 1 {
+		return nil, ErrTooManyArgs
+	} else if len(args) < 1 {
+		*hashFn = crypto.SHA256
+	} else {
+		*hashFn = args[0]
+	}
+	if !hashFn.Available() {
+		return nil, ErrHashFnNotAvailable
+	}
+	return hashFn, nil
+}
```

Comment:
This can be simplified.
```suggestion
func ValidateHashFunction(hashFn crypto.Hash) crypto.Hash {
	if hashFn.Available() {
		return hashFn
	}
	return crypto.SHA256
}
```